packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -98,25 +99,83 @@ export const Block = Node.create<IBlock>({
     };
   },
 
-  // TODO: should we parse <li>, <ol>, <h1>, etc?
   parseHTML() {
     return [
+      // For all content copied from within the editor.
       {
         tag: "div",
+        getAttrs: (element) => {
+          if (typeof element === "string") {
+            return false;
+          }
+
+          if (element.getAttribute("node") === "block") {
+            // Null means the element matches, but we don't want to add any attributes to the node.
+            return null;
+          }
+
+          return false;
+        }
+      },
+      // For headings & paragraphs copied from outside the editor.
+      {
+        tag: "p",
+        priority: 100
+      },
+      {
+        tag: "h1",
+        attrs: {headingType: 1}
+      },
+      {
+        tag: "h2",
+        attrs: {headingType: 2}
+      },
+      {
+        tag: "h3",
+        attrs: {headingType: 3}
+      },
+      // For lists copied from outside the editor.
+      {
+        tag: "li",
+        getAttrs: (element: HTMLElement | string) => {
+          if (typeof element === "string") {
+            return false;
+          }
+
+          let ancestor = element.parentElement;
+
+          if (ancestor === null) {
+            return false;
+          }
+
+          while (ancestor.tagName !== "B") {
```

Comment:
this while doesn't make sense? ancestor is never changed?

Reply:
Whoops, it was supposed to traverse up the DOM tree until it finds an ancestor that's a list. I guess it's not needed either way since list items always have a list as a parent.